utils/plots.py

```diff
@@ -0,0 +1,152 @@
+    ):
+        top_predictions = []
+        
+        # Get top N predictions for each miRNA
+        grouped = df.groupby('miRNA')
+        for name, group in grouped:
+            top_number_of_preds = group.nsmallest(num_of_top_preds, label)
+            top_predictions.append(top_number_of_preds['fold_change'])
+        
+        # Concatenate the top predictions into a single array
+        top_predictions = pd.concat(top_predictions)
+        #plot FCs of top N predictions
+        plot_lines_with_median(top_predictions, label, color)
+    
+    plt.xlim(-2, 2)
+    plt.xlabel('Log2 Fold Change (LFC)')
+    plt.ylabel('Cumulative Density Function (CDF)')
+    plt.title(f'CDF of Top {num_of_top_preds} Predicted Targets per miRNA - {title_sufix}')
+    plt.legend(loc='lower right')
+    plt.gca().spines['top'].set_visible(False)
+    plt.gca().spines['right'].set_visible(False)
+    plt.gca().spines['bottom'].set_color('black')
+    plt.gca().spines['left'].set_color('black')
+    plt.gca().yaxis.set_ticks_position('left')
+    plt.gca().xaxis.set_ticks_position('bottom')
+    plt.grid(False)
+    plt.show()
+
+
+def plot_correlation(y_true, y_pred, title_sufix):
+    plt.figure(figsize=(5, 5),dpi=300)
+    xlow, xhigh, ylow, yhigh = -0.5,0.5,-0.5,0.5
+    hexplot = sns.jointplot(
+        x=y_true,
+        y=y_pred,
+        kind='hist',
+        color='darkred',
+    )
+    sns.regplot(
+        x=y_true, 
+        y=y_pred, 
+        scatter_kws=dict(alpha=0),
+        color='brown'
+    )
+    plt.text(
+        0.7, 
+        0.9, 
+        f"pearson, r = {pearsonr(y_true, y_pred)[0]:.2f}",
+        # f"r = {r_regression(y_true, y_pred):.2f}",
+        ha='center'
+    )
+    plt.text(
+        0.7, 
+        0.8, 
+        f"r^2 = {r2_score(y_true, y_pred):.2f}",
+        ha='center'
+    )
+    plt.text(
+        0.7, 
+        0.7, 
+        f"spearman = {spearmanr(y_true, y_pred)[0]:.2f}",
+        # f"r = {r_regression(y_true, y_pred):.2f}",
+        ha='center'
+    )
+
+    plt.xlim(-1,1)
+    plt.ylim(-1,1)
+    # plt.grid(b=None)
+    plt.grid(False)
+    plt.xlabel('Log2 Fold Change (LFC)')
+    plt.ylabel('Predictions')
+    # plt.title(f'Correlation of predictions and LFC - {title_sufix}', y=1.5)
+    plt.text(0, 1.5, f'Correlation of predictions and LFC - {title_sufix}',
+         horizontalalignment='center')
+    plt.gca().spines['bottom'].set_color('black')
+    plt.gca().spines['left'].set_color('black')
+    plt.gca().yaxis.set_ticks_position('left')
+    plt.gca().xaxis.set_ticks_position('bottom')
+    
+    
+def plot_feature_importance(feature_names, feature_importances):
+    # Create a pandas DataFrame for seaborn
+    importance_df = pd.DataFrame({
+        'Feature Names': feature_names,
+        'Importances': feature_importances
+    })
+
+    # Sort by importance
+    importance_df = importance_df.sort_values('Importances', ascending=False)
+
+    # Create the plot
+    plt.figure(figsize=(5, 4),dpi=150)
```

Comment:
For publication, you should do dpi of at least 300

Reply:
done